### ISSUE TITLE
CVAT doesn't export instance annotations in the correct format

### ISSUE THREAD

**Issue by davodogster** [openvinotoolkit/datumaro#28]
Hi there. I'm a big fan of using the openvino model assisted annotation. So I need to train a MaskRCNN and convert it to openvino then upload the model to cvat (have done it for faster rcnn). But to train MaskRCNN you need a TFrecord file and for that you need the XMLS( in the correct format) with the bounding box coordinates of the polygons and also the PNG masks. When I dump annotations in the sementation mask format I don't get the XMLS. When I export in other format coco it has a json file which has the bbox co-ords but that is not what's needed.. We need an xml per image not one big Json file in the wrong format.

CVAT doesn't export the write annotation files needed to (convert to tfrecord) and train maskRCNN.

I'm Running CVAT in docker. Using a slightly older version - just before serverless functions were introduced. 

Cheers,
Sam

**Comment by nmanovic** [comment]
@zhiltsov-max , @davodogster , I have moved the issue into Datumaro repo. I believe the use case should be supported somehow inside the framework first.

**Comment by zhiltsov-max** [comment]
There is an option `--save-masks` in `tf_detection_api` format of Datumaro, but in CVAT it is not used by default. How do you think, would it be adequate, if it was turned on by default (so the polygons were converted to masks, like in `mask` format of CVAT)? Maybe it should be another format, or an export option?

**Comment by davodogster** [comment]
The masks are exported (voc segmentation) as PNG but they miss classes when there is more than one class as mentioned in the other issue from someone else (maybe it was fixed in the recent version of CVAT). The coco instance annotations (json) work for training TF1OD maskrcnn if you first run create_coco_tf_record.py to generated the sharded tfrecords. I have got it running. I had to use this helpful script to split the instances_default.json into training and validation https://github.com/akarazniewicz/cocosplit. It would be great if cvat exported pascal voc xmls with the bbox cordinates of the objects so that users can use a modified version of create_pet_tf_record.py to generate. 2 ways to train MaskRCNN tensorflow - 1. PNGs + Pascal bbox XMLS, 2. instances_default.json. Both get converted to tfrecord outside CVAT. even if you could do it in CVAT.. it wouldn't address the train-test split problem :)

**Comment by davodogster** [comment]
@zhiltsov-max Hi Max, actually the Pascal VOC PNG Mask + XML per image is the much preferred annotation option for instance segmentation! This is because users need to merge training data from different tasks in cvat which is not possible if you just have one coco json file for each CVAT task (as opposed to each image). I annotate a task, train a model, upload the model to CVAT and apply it to a new task, then download the annotations then merge annotations from task 1 and task 2 (much easier if it's per image annotations instead of per task). Hope this makes sense. Sam

**Comment by zhiltsov-max** [comment]
Take a look at [Datumaro documentation ("transform", "merge", "export", "add and remove data" sections)](https://github.com/openvinotoolkit/datumaro/blob/develop/docs/user_manual.md), I think, you might find something interesting there - particularly, about merging and splitting of datasets. For example, coco annotations can be easily merged.

What you are describing is just a full PASCAL VOC format - with xmls and masks, so what about using:
```
# export task as Datumaro project
# the project will have polygons, so we create a project with boxes and a project with masks

datum project transform -t polygons_to_masks -o masks_proj
datum project export -p masks_proj -f voc [-- --save-images]

datum project transform -t shapes_to_boxes -o boxes_proj
datum project export -p boxes_proj -f voc
```

**Comment by davodogster** [comment]
@zhiltsov-max Thanks so much for this info! It looks promising so I will look into it. And I'm very happy to hear that task merging will be available in CVAT soon.

**Comment by davodogster** [comment]
@zhiltsov-max The import command just created an empty directory called train_proj with nothing in it 
![image](https://user-images.githubusercontent.com/46079516/94639610-111c0400-0339-11eb-8483-3d9f6592fe67.png)

**Comment by zhiltsov-max** [comment]
There should be a `.datumaro` subdirectory, which is hidden by default.

**Comment by davodogster** [comment]
@zhiltsov-max I think datumaro is a great concept.. but it needs more work - your basic functions fail for me.

Here it is (raw, looks good): 
![image](https://user-images.githubusercontent.com/46079516/95153643-6601c380-07ec-11eb-8845-8a77f5cdffc4.png)

I exported the polygons as a Datumaro dataset, ran the transform command which just created a new (failed) default.json
![image](https://user-images.githubusercontent.com/46079516/95153619-55e9e400-07ec-11eb-88ea-01f7a76b8a8e.png)

failed results:
![image](https://user-images.githubusercontent.com/46079516/95153761-bc6f0200-07ec-11eb-926d-e47a80c60d53.png)


Please help - Cheers, Sam

**Comment by zhiltsov-max** [comment]
Run `export ... -- --label-map=source`.

**Comment by davodogster** [comment]
@zhiltsov-max Hi max, thank you for that additional code. It works if I run `datum export -p masks_proj -f voc -- --label-map=source`, I have npw exported a datumaro polygon dataset from CVAT and successfully converted it to coco json, png masks, tf record, and converted it to voc boxes, and then boxes to yolo. datum project transform is an interesting command, it seems to create a strange intermediate dataset which can then be converted to a proper dataset. COCO and TF instance format don't require the --label-map param.

**Comment by zhiltsov-max** [comment]
Closing, the answer is in https://github.com/openvinotoolkit/datumaro/issues/28#issuecomment-700514967
